lgn-provers/src/provers/v0/groth16/mod.rs

```diff
@@ -9,12 +9,16 @@ mod prover;
 mod task;
 
 #[allow(unused_variables)]
+#[allow(clippy::too_many_arguments)]
 pub fn create_prover(
     url: &str,
     dir: &str,
     circuit_file: &str,
+    circuit_file_checksum: &str,
```

Comment:
Can't we simply do the checksum over the whole directory maybe ? Simpler ? It's best to be atomic on this stuff, we don't want to have different versions of the sub-parameters living the same place.